### PR TITLE
Fix Remove redundant components named exports in favor of the default ones.

### LINKED ISSUE #4253: Components exported twice
Some components are exported twice, as default and as not default, I don't no if it's the idea.

- ArrayInput
- DatagridCell
- DateInput
- DateTimeInput
- FileField
- ImageField
- Labeled
- LoadingIndicator
- ReferenceArrayField
- ReferenceInput
- TextInput

- List, Edit, Show (exports its `useStyles` hooks)

### PR DESCRIPTION
**Pull request by WiXSL**
Fixes #4253.

### CODE REVIEW

**Comment by WiXSL** [comment]
Some components test failed when importing the default version which is a `recompos.pure` or `recompos.shouldComponentUpdate` version of them.

May be I could change tat for `React.memo` in another PR.

**Comment by WiXSL** [comment]
Done!

File: packages/ra-ui-materialui/src/list/DatagridHeaderCell.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const useStyles = makeStyles(
     { name: 'RaDatagridHeaderCell' }
 );
 
-export const DatagridHeaderCell = ({
+const DatagridHeaderCell = ({
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/DatagridHeaderCell.js]: I think this one has a HOC and should still be exported

File: packages/ra-ui-materialui/src/list/FilterForm.js [hunk 2]
```diff
@@ -82,7 +82,7 @@ const sanitizeRestProps = ({
     ...props
 }) => props;
 
-export const FilterForm = ({
+const FilterForm = ({
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/FilterForm.js]: same with this one

Comment (WiXSL) [reply packages/ra-ui-materialui/src/list/FilterForm.js]: Done.

File: packages/ra-ui-materialui/src/list/DatagridHeaderCell.js (hunk 1 above)
Comment (WiXSL) [reply packages/ra-ui-materialui/src/list/DatagridHeaderCell.js]: Done.

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**